packages/glimmer-runtime/lib/builder.ts

```diff
@@ -279,6 +279,14 @@ export class ElementStack {
     return fragmentBounds;
   }
 
+  insertTextBefore(nextSibling: Node, text: string): Fragment {
```

Comment:
(to be clear, this is great work – I don't mean I don't like the implementation, I just mean I'm sad that implementing this feature forces us to pollute the common path with a lot of stuff that handles the edge cases, and would like to eventually figure out how to not do that)